src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/CodeGeneration/DesignTimeNodeWriter.cs

```diff
@@ -209,38 +194,33 @@ public override void WriteCSharpExpressionAttributeValue(CodeRenderingContext co
 
     public override void WriteCSharpCodeAttributeValue(CodeRenderingContext context, CSharpCodeAttributeValueIntermediateNode node)
     {
+        var writer = context.CodeWriter;
+
         foreach (var child in node.Children)
         {
             if (child is CSharpIntermediateToken token)
             {
-                IDisposable linePragmaScope = null;
-                var isWhitespaceStatement = string.IsNullOrWhiteSpace(token.Content);
-
-                if (token.Source != null)
-                {
-                    if (!isWhitespaceStatement)
-                    {
-                        linePragmaScope = context.CodeWriter.BuildLinePragma(token.Source.Value, context);
-                    }
-
-                    context.CodeWriter.WritePadding(0, token.Source.Value, context);
-                }
-                else if (isWhitespaceStatement)
+                if (token.Content.IsNullOrWhiteSpace())
                 {
-                    // Don't write whitespace if there is no line mapping for it.
                     continue;
```

Comment:
>continue; [](http://example.com/codeflow?start=20&length=9)

Just making sure, this is an intentional behavior change when there is a token.Source, right?

Reply:
Yeah, I think I got this refactoring wrong. I'm not sure all of the original behavior actually runs on real code, but I'll fix this up to match:

- If source is null and the token is whitespace...
    - Continue looping
- If source is null and token is non-whitespace...
    - Write the token as a single line.
- If source is not null and the token is non-whitespace...
    - Write a line pragma
    - Write padding for the token
    - Add a source mapping
    - Write the token content
- If source is not null and the token is whitespace
    - Write padding for the token
    - Add a source mapping
    - Write the token as a single line

I suspect that we really don't ever produce all-whitespace CSharpIntermediateTokens with non-null Source.